index.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+/* eslint-disable no-console */
+
+class Datastore {
+    /**
+     * Constructor for Datastore
+     * @method constructor
+     * @param  {Object}    config Configuration
+     * @return {Datastore}
+     */
+    constructor(config) {
+        this.config = config;
+    }
+
+    /**
+     * Reload configuration
+     * @method configure
+     * @param  {Object}     config      Configuration
+     * @param  {Function}   [callback]
+     */
+    configure(config) {
+        this.config = config;
+    }
+
+    /**
+     * Get a single record given an id
+     * @method get
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.table      table name
+     * @param  {Object}   config.params     query params
+     * @param  {Object}   config.params.id  Unique id
+     * @param  {Function} callback
+     */
+    get() {
+        console.error('get is not implemented');
```

Comment:
I want to avoid bringing in any more dependencies until there is a good reason for someone to use this class. Since any implementers have to override everything for now, there's no point in brining anything in.